Subject: Debian paths and commands
Last-Update: 2010-04-11

User pathnames and commands specific to the Debian platform.
TODO: _PATH_DEFPATH_ROOT does not appear to be used (see cron.c)

Bug-Debian: http://bugs.debian.org/482284
Index: patched/pathnames.h
===================================================================
--- patched.orig/pathnames.h	2010-05-06 18:18:22.673435429 +0200
+++ patched/pathnames.h	2010-05-06 18:18:46.636435718 +0200
@@ -28,7 +28,7 @@
 			 * to; SPOOL_DIR, ALLOW_FILE, DENY_FILE, and LOG_FILE
 			 * are all relative to this directory.
 			 */
-#define CRONDIR		"/var/cron"
+#define CRONDIR		"/var/spool/cron"
 #endif
 
 			/* SPOOLDIR is where the crontabs live.
@@ -39,7 +39,7 @@
 			 * newer than they were last time around (or which
 			 * didn't exist last time around...)
 			 */
-#define SPOOL_DIR	"tabs"
+#define SPOOL_DIR	"crontabs"
 
 			/* undefining these turns off their features.  note
 			 * that ALLOW_FILE and DENY_FILE must both be defined
@@ -47,8 +47,13 @@
 			 * LOG_FILE or SYSLOG is defined, we don't log.  If
 			 * both are defined, we log both ways.
 			 */
+#ifdef DEBIAN
+#define	ALLOW_FILE	"/etc/cron.allow"		/*-*/
+#define DENY_FILE	"/etc/cron.deny"		/*-*/
+#else
 #define	ALLOW_FILE	"allow"		/*-*/
 #define DENY_FILE	"deny"		/*-*/
+#endif
 #define LOG_FILE	"log"		/*-*/
 
 			/* where should the daemon stick its PID?
@@ -58,7 +63,7 @@
 #else
 # define PIDDIR "/etc/"
 #endif
-#define PIDFILE		"%scron.pid"
+#define PIDFILE		"%scrond.pid"
 
 			/* 4.3BSD-style crontab */
 #define SYSCRONTAB	"/etc/crontab"
@@ -66,7 +71,9 @@
 			/* what editor to use if no EDITOR or VISUAL
 			 * environment variable specified.
 			 */
-#if defined(_PATH_VI)
+#if defined(DEBIAN)
+# define EDITOR "/usr/bin/sensible-editor"
+#elif defined(_PATH_VI)
 # define EDITOR _PATH_VI
 #else
 # define EDITOR "/usr/ucb/vi"
@@ -79,3 +86,7 @@
 #ifndef _PATH_DEFPATH
 # define _PATH_DEFPATH "/usr/bin:/bin"
 #endif
+
+#ifndef _PATH_DEFPATH_ROOT
+# define _PATH_DEFPATH_ROOT "/usr/sbin:/usr/bin:/sbin:/bin"
+#endif
Index: patched/misc.c
===================================================================
--- patched.orig/misc.c	2010-05-06 18:18:44.201560954 +0200
+++ patched/misc.c	2010-05-06 18:18:46.637435837 +0200
@@ -234,7 +234,7 @@
 }
 
 
-/* acquire_daemonlock() - write our PID into /etc/cron.pid, unless
+/* acquire_daemonlock() - write our PID into /etc/crond.pid, unless
  *	another daemon is already running, which we detect here.
  *
  * note: main() calls us twice; once before forking, once after.
